nfts/src/tests/extrinsics.rs

```diff
@@ -358,16 +358,15 @@ mod delegate {
 	#[test]
 	fn delegate() {
 		ExtBuilder::new_build(vec![(ALICE, 100)]).execute_with(|| {
-			let mut nft = NFTs::data(ALICE_NFT_ID).unwrap();
 			let viewer = Some(BOB);
 
 			// Delegating nft to another account
-			let ok = NFTs::delegate(origin(ALICE), ALICE_NFT_ID, viewer.clone());
+			let ok = NFTs::delegate(origin(ALICE), ALICE_NFT_ID, viewer);
 			assert_ok!(ok);
 
 			// Final state checks
-			nft.viewer = viewer.clone();
-			assert_eq!(NFTs::data(ALICE_NFT_ID), Some(nft));
+			assert_eq!(NFTs::data(ALICE_NFT_ID).as_ref().unwrap().is_delegated, true);
```

Comment:
In the previous code I intentionally check that the whole NFT structure is in the right form together with the viewer field.
In your code you just check if the is_delegated flag is set to true, without checking if the rest of the object is in the correct state.

Reply:
Done